test-suite/testkit/src/api.rs

```diff
@@ -261,9 +283,25 @@ where
             builder = modifier(builder);
         }
         let response = builder.send().expect("Unable to send request");
+        Self::verify_headers(self.expected_headers, &response);
         Self::response_to_api_result(response)
     }
 
+    // Checks that response contains headers expected by the request author.
+    fn verify_headers(expected_headers: HashMap<String, String>, response: &Response) {
+        let headers = response.headers();
+        for (header, expected_value) in expected_headers.iter() {
+            let header_value = headers.get(header).unwrap_or_else(|| {
+                panic!(
+                    "Response {:?} was expected to have header {}, but it isn't present",
+                    response, header
+                );
+            });
+
+            assert_eq!(header_value, expected_value);
```

Comment:
Nit: It could make sense to add a message to `assert_eq` macro for additional context, something like `"Unexpected value of response header {}"`.